### ISSUE TITLE
Improve the accessibility of the "unavailable message" on authenticators

### ISSUE THREAD

**Issue by ScopeyNZ** [silverstripe/silverstripe-mfa#77]
### Background
Currently when authenticators are unavailable they show as greyed out with some text overlayed indicating the reason why the authenticator is unavailable. 

This isn't very accessible:

![image](https://user-images.githubusercontent.com/3260989/55913365-98068980-5c38-11e9-884c-a5c40c4282a6.png)

### Changes required

New designs are suggested here:
https://projects.invisionapp.com/share/3PNSKZQYBJZ#/screens/322769156

**Comment by newleeland** [comment]
Hey @ScopeyNZ 

With the 75% transparent white over the darkest part of that graphic (the dark grey) produces a relative colour of: **#CBCDD1** 
When placed under text of **#43536D** produces a contrast level of 4.89 which passes WCAG AA contrast levels. We could increase the overlay from 75% to 80% if you feel the text in the foreground is not legible enough.

OR Are you worried about the text that the overlay is on top of?

**Comment by robbieaverill** [comment]
@newleeland we're building configurable and extensible code here, so people could create their own authenticators with their own graphics. It's probably worth considering the absolute worst case scenario image someone could use and making the mask sufficient for AA against it. Would a full black image be the worst case? If so, it equates to a #c6c6c6 ground with #546780 text, which fails AA for small text

**Comment by ScopeyNZ** [comment]
I think we just shouldn't have the text over the image tbh. I think it's just way too much going on there. To be honest I didn't run the contrast checker over it - I think that it's not particularly relevant in this section.

Admittedly I'm not as well versed in accessibility standards around this - but I think it's pretty hard to see the text personally - it might just be me though.

**Comment by newleeland** [comment]
@robbieaverill @ScopeyNZ 

With Black #000 graphics under 80% #FFF, the colour corresponds to **#CCC** when contrasted to **#43536D** is 4.85:1

The body-text colour we are using is #43536D and not #546780 (i.e. header colour)

**Comment by brynwhyman** [comment]
We've since talked about moving the 'unavailable' messaging to replace the description text, so it's not over the graphic.

**Comment by newleeland** [comment]
@brynwhyman I've tried moving the "unsupported" message down, replacing the description of the Security Key however it was not clear why the method isn't selectable, until the user read the description.
Alternatively, I've tried moving the text down and overlaying the descriptive text. However, this was not that much more legible than having it on top with the imagery.

This is potentially what we can do.
https://projects.invisionapp.com/share/3PNSKZQYBJZ#/screens/322769156

**Comment by robbieaverill** [comment]
@newleeland @clarkepaul I think I remember talking about this in a grooming session, and we suggested that we could replace the method's description with the unsupported message and make it disabled - is that still an option, or should I implement what's in those Invision designs?

**Comment by Cheddam** [comment]
Replacing the description with the unsupported message is kind of mixing UI purposes - and would be less visible. If we combined it with greying out the icon and border, it'd probably suffice though. I'm not sure about the overlay in the new design either to be honest.

**Comment by robbieaverill** [comment]
Blocked awaiting feedback

**Comment by newleeland** [comment]
I've tried a couple of alternative methods to make the text more legible, to no avail. i.e.
- Replacing description text with unavailable message. Which was not very clear
- Overlay the method description so message is on the fore front. But this was still very hard to read

@clarkepaul and myself settle on this:
https://projects.invisionapp.com/share/3PNSKZQYBJZ#/screens/322769156

**Comment by newleeland** [comment]
> @newleeland @clarkepaul I think I remember talking about this in a grooming session, and we suggested that we could replace the method's description with the unsupported message and make it disabled - is that still an option, or should I implement what's in those Invision designs?

We tried this method, and it wasn't very apparent that it was unavailable and why it was unselectable until the user read the the description. Hence we came up with the alternative below:

>@brynwhyman I've tried moving the "unsupported" message down, replacing the description of the Security Key however it was not clear why the method isn't selectable, until the user read the description.
Alternatively, I've tried moving the text down and overlaying the descriptive text. However, this was not that much more legible than having it on top with the imagery.
This is potentially what we can do.
https://projects.invisionapp.com/share/3PNSKZQYBJZ#/screens/322769156

**Comment by robbieaverill** [comment]
Cool, so that's the designs we're going with? Moving back to ready

**Comment by newleeland** [comment]
Yes! it's been approved from the design end.
